### PR TITLE
@W-9531390@: Smoke tests now run against tarballs.

### PR DESCRIPTION
**Pull request by jfeingold35**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -145,34 +151,107 @@ jobs:
...
+
+      - run:
+          name: "Install sfdx-cli"
+          command: npm install -g sfdx-cli
+
+      # Build and install our tarball.
+      - run:
+          name: "Build and install tarball"
+          command: |
+            yarn pack --filename tarball.tgz
```
Comment (jbartolotta-sfdc) [inline .circleci/config.yml:181]: I compared `npm package` and `yarn pack`. I believe we should use `npm package`. This was the difference when i exploded the tars into directories and compared them.

```
Only in from-yarn/package: github-actions
Only in from-npm/package: NOTICE
```

File: package.json [hunk 2]
```diff
@@ -108,7 +108,7 @@
 	"repository": "forcedotcom/sfdx-scanner",
 	"scripts": {
 		"build": "./gradlew build",
-		"prepack": "rm -rf lib && tsc -b && oclif-dev manifest && oclif-dev readme",
+		"prepack": "rm -rf lib && npx tsc -b && npx oclif-dev manifest && npx oclif-dev readme",
```
Comment (jbartolotta-sfdc) [inline package.json:111]: Why is this needed?

File: .circleci/config.yml [hunk 3]
```diff
@@ -145,34 +151,107 @@ jobs:
...
           at: C:\repo
 
+      # Upgrade node to whatever version we need.
       - run:
-          name: "Upgrade node to LTS"
+          name: "Upgrade node"
           shell: bash.exe
-          command: .circleci/windows-node-upgrade.sh lts
+          command: |
+            DESIRED_VERSION="current" && [[ << parameters.lts >> == true ]] && DESIRED_VERSION="lts"
```
Comment (jbartolotta-sfdc) [inline .circleci/config.yml:223]: Can this use a consistent name with the upgrade script? The upgrade script uses `WANTED_VERSION`

File: .circleci/config.yml [hunk 4]
```diff
@@ -145,34 +151,107 @@ jobs:
...
+          root: ~/repo
+          paths: .
+
+      # Create a directory for the test output.
+      - run: mkdir test-results
+
+      # Run the smoke tests. If none of them explode, we can reasonably assume that the plugin is safe.
+      - run:
+          name: "Smoke Tests"
+          command: smoke-tests/smoke-test.sh sfdx
```
Comment (jbartolotta-sfdc) [inline .circleci/config.yml:196]: This file shows up as `Empty file.` for me on this review. Any idea why?

File: .circleci/config.yml (hunk 1 above)
Comment (jfeingold35) [reply .circleci/config.yml:181]: I was using `yarn pack` because it let me specify the name of the outfile. I'll switch to `npm pack` and add a `grep`, which should be fine.

File: .circleci/config.yml (hunk 4 above)
Comment (jfeingold35) [reply .circleci/config.yml:196]: I didn't change the contents of the file; I just changed its default permissions so the CI can execute it, which I was seeing it struggle to do.

File: .circleci/config.yml (hunk 3 above)
Comment (jfeingold35) [reply .circleci/config.yml:223]: Will do.

File: package.json (hunk 2 above)
Comment (jfeingold35) [reply package.json:111]: I sincerely have no idea. For some reason, when I ran the scripts on Linux, they worked fine, but only worked on Windows when I used `npx` to run the local versions instead of a global version.
I had to add `npx` to the `test` task when we first added the Windows tests, for the same reason.

Comment (jbartolotta-sfdc) [reply package.json:111]: We should be able to remove this since we are no longer building on Windows.

Comment (jfeingold35) [reply package.json:111]: Good point. I'll do that.

**Comment by jfeingold35** [comment]
This was superseded by #489
